lib/app/service/storage_service/browser_bookmarks_storage_service.dart

```diff
@@ -70,12 +70,26 @@ class BrowserBookmarksStorageService extends AbstractStorageService {
   }
 
   /// Clear browser bookmarks
-  Future<void> clearBrowserBookmarks() async {
+  Future<void> clearBrowserBookmarks({
+    bool needUndo = true,
+  }) async {
     await _storage.delete(
       _browserBookmarksKey,
       domain: _browserBookmarksDomain,
     );
 
+    final savedbrowserBookmarks = browserBookmarks;
```

Comment:
I think, you should save this variable before doing `delete` action, isn't it?

Reply:
And it looks like we should add same logic for deleting single bookmark

Reply:
Yeah, thank you very much! Fixed.

Reply:
When deleting single bookmark we have `item` object that we can restore when undo'ing.